### PR TITLE
[RFC] Change return type to bool on fileio.c functions

### PR DESCRIPTION
**Pull request by war1025**

### CODE REVIEW

File: src/nvim/fileio.c [hunk 1]
```diff
@@ -4082,19 +4091,21 @@ static int buf_write_bytes(struct bw_info *ip)
...
+///
+/// @param         c     The character to convert
+/// @param[in,out] pp    The pointer to store the result at
+/// @param         flags The FIO_ flags to use
+///
+/// @return true for an error, false when it's OK.
+static bool ucs2bytes(
+    unsigned c,
+    char_u **pp,
+    int flags
```
Comment (cacplate) [inline src/nvim/fileio.c]: This doesn't follow the style guide. It should be on same line or aligned on parenthese if too long.

File: src/nvim/fileio.c [hunk 2]
```diff
@@ -5352,10 +5371,12 @@ static int au_find_group(char_u *name)
...
 
-/*
- * Return TRUE if augroup "name" exists.
- */
-int au_has_group(char_u *name)
+/// Return true if augroup "name" exists.
+///
+/// @param name The augroup name
+///
+/// @return true if augroup "name" exists.
```
Comment (cacplate) [inline src/nvim/fileio.c]: This is redundant. I don't know what is the best approach but I recall reading an issue about this, can't find it though.

File: src/nvim/fileio.c [hunk 3]
```diff
@@ -5472,21 +5493,25 @@ find_end_event (
...
 
-/*
- * Return TRUE if "event" is included in 'eventignore'.
- */
-static int event_ignored(event_T event)
+/// Return true if "event" is included in 'eventignore'.
+///
+/// @param event The event to check
+///
+/// @return true if `event` is included in `eventignore`.
```
Comment (cacplate) [inline src/nvim/fileio.c]: same.

File: src/nvim/fileio.c [hunk 4]
```diff
@@ -6191,55 +6218,78 @@ aucmd_restbuf (
...
+bool apply_autocmds(
     event_T event,
-    char_u *fname,         /* NULL or empty means use actual file name */
-    char_u *fname_io,      /* fname to use for <afile> on cmdline */
-    int force,                  /* when TRUE, ignore autocmd_busy */
-    buf_T *buf           /* buffer for <abuf> */
+    char_u *fname,
+    char_u *fname_io,
+    int force,
+    buf_T *buf
```
Comment (cacplate) [inline src/nvim/fileio.c]: same. Should be aligned with parenthese.

File: src/nvim/fileio.c [hunk 5]
```diff
@@ -6271,75 +6321,88 @@ int trigger_cursorhold(void)
...
 }
 
-static int 
-apply_autocmds_group (
+/// Execute autocommands for "event" and file name "fname".
+///
+/// @param event    The event that occured
+/// @param fname    The filename, NULL or empty means use actual file name
+/// @param fname_io The filename to use for <afile> on cmdline,
+//                  NULL means use `fname`.
```
Comment (cacplate) [inline src/nvim/fileio.c]: `///`

File: src/nvim/fileio.c [hunk 6]
```diff
@@ -3893,7 +3896,13 @@ static int check_mtime(buf_T *buf, FileInfo *file_info)
...
 }
 
-static int time_differs(long t1, long t2)
+/// Return true if the times differ
+///
+/// @param t1 The first time
+/// @param t2 The second time
+///
+/// @return true if the times differ
+static bool time_differs(long t1, long t2)
```
Comment (ZyX-I) [inline src/nvim/fileio.c]: This should have FUNC_ATTR_CONST.

File: src/nvim/fileio.c [hunk 7]
```diff
@@ -1992,13 +1992,15 @@ readfile (
...
-static int is_dev_fd_file(char_u *fname)
+/// Returns true if the file name argument is of the form "/dev/fd/\d\+",
+/// which is the name of files used for process substitution output by
+/// some shells on some operating systems, e.g., bash on SunOS.
+/// Do not accept "/dev/fd/[012]", opening these may hang Vim.
+///
+/// @param fname The file name to check
+///
+/// @return true if the file name argument is of the form "/dev/fd/\d\+",
+static bool is_dev_fd_file(char_u *fname)
```
Comment (ZyX-I) [inline src/nvim/fileio.c]: FUNC_ATTR_NONNULL_ALL

Comment (ZyX-I) [inline src/nvim/fileio.c]: @justinmk Actually, should such PRs also modify the function attributes if they modify a function signature?

Comment (justinmk) [inline src/nvim/fileio.c]: As part of lint? That would be good. The PRs by @war1025  may need to be adopted by someone though.

Comment (cacplate) [inline src/nvim/fileio.c]: I can take them if needed. How can I do it?

On Wed, Jan 06, 2016 at 6:59 pm, Justin M. Keyes < notifications@github.com [notifications@github.com] > wrote:
In src/nvim/fileio.c
[https://github.com/neovim/neovim/pull/3698#discussion_r49026859] :

> -/*
> - * Returns TRUE if the file name argument is of the form "/dev/fd/\d\+",
> - * which is the name of files used for process substitution output by
> - * some shells on some operating systems, e.g., bash on SunOS.
> - * Do not accept "/dev/fd/[012]", opening these may hang Vim.
> - */
> -static int is_dev_fd_file(char_u *fname)
> +/// Returns true if the file name argument is of the form "/dev/fd/\d\+",
> +/// which is the name of files used for process substitution output by
> +/// some shells on some operating systems, e.g., bash on SunOS.
> +/// Do not accept "/dev/fd/[012]", opening these may hang Vim.
> +///
> +/// @param fname The file name to check
> +///
> +/// @return true if the file name argument is of the form "/dev/fd/\d\+",
> +static bool is_dev_fd_file(char_u *fname)


As part of lint? That would be good. The PRs by @war1025 [https://github.com/war1025] may need to be adopted by someone though.

Comment (ZyX-I) [inline src/nvim/fileio.c]: @cacplate AFAIK the only way to take others’ PR is make one of the necessary access close them and make your PR (not necessary in this order). Changes can be pulled from `refs/pull/$nr/head`: e.g. I have an alias `fpr = "!sh -c 'nr=\"$1\" ; shift ; git fetch upstream refs/pull/$nr/head:pr-$nr \"$@\"' -"` that when `git fpr 42` is run it creates local branch `pr-42` out of PR number 42.

Comment (ZyX-I) [inline src/nvim/fileio.c]: @cacplate AFAIK the only way to take others’ PR is make somebody with the necessary permissions close them and make your PR (not necessary in this order). Changes can be pulled from `refs/pull/$nr/head`: e.g. I have an alias `fpr = "!sh -c 'nr=\"$1\" ; shift ; git fetch upstream refs/pull/$nr/head:pr-$nr \"$@\"' -"` that when `git fpr 42` is run it creates local branch `pr-42` out of PR number 42.

**Pull request closed without merging**